### DISCUSSION TITLE
C++ version..?

### DISCUSSION THREAD

**Discussion by mkoenings** [cycfi/q#69] [Q&A] [answered]
Thank you for this awesome library, it does exactly what I  need for rmy crazy guitar project. I need super fast pitch detection to calculate a phase correction (delay) for a sustain/feedback effect device.
I want to use the library on small MCUs like the Teensy 4.1 , but  the tool chains do not support some of  the fine C++ 20 features that you use in your code (i.e. concepts) 
How can I work around this ? I am not a C++ expert, I just fell into this "rabbit hole"  on my journey trying to make the  project idea work. Would it be possible to create a "downgrade"  ( C++14) version of this code ? How do you do it on your hardware ?
Cheers and thanks for the brilliant library,
Michael

**Comment by djowel** [comment, marked as answer]
> Thank you for this awesome library, it does exactly what I need for rmy crazy guitar project. I need super fast pitch detection to calculate a phase correction (delay) for a sustain/feedback effect device. I want to use the library on small MCUs like the Teensy 4.1 , but the tool chains do not support some of the fine C++ 20 features that you use in your code (i.e. concepts) How can I work around this ? I am not a C++ expert, I just fell into this "rabbit hole" on my journey trying to make the project idea work. Would it be possible to create a "downgrade" ( C++14) version of this code ? How do you do it on your hardware ? Cheers and thanks for the brilliant library, Michael

I'm using the Q lib in an embedded system (STM32F4 and STM32H7). You should just update your toolchain to the latest compiler.
